### DISCUSSION TITLE
Handling of problematic samples

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#20] [General]
Regarding handling of "failed" samples - do we have specific thresholds that could be applied?

It is technically possible to filter out samples along the workflow based on the content of e.g. reports and such. As an example, see GABI where the confindR report is read (using Groovy) to branch samples based on whether or not they have failed.

[    CONFINDR.out.report.map { m, r ->](https://github.com/marchoeppner/gabi/blob/38fe92e6e1d49d2960be86bdeedd4c5c97e36cfd/subworkflows/contamination/main.nf#L24)

**Comment by gregdenay** [comment]
I was not really thinking of QC thresholds for this, more like if a sample it so bad that DADA exits with an error or there is no BLAST hits.
maybe we could handle this with a filter for less than e.g. 500 reads? Alternatively we could check the exit status/output (depending on the tool) of the critical steps (e.g. no reads after trimming, DADA error, BLAST empty...). The point would be to avoid crashing the whole run for one bad sample AND report on QC even if a sample doesn't run through to the end.
What do you think is the most robust and practicable?

**Reply by marchoeppner** [reply]
Right, as you said, that will depend. 

I can absolutely put in a filter after trimming to stop samples that fall below a very obvious threshold, for starters. And then we have to take it from there and decide on a case by case bases how specific fatal errors might best be "caught".

**Reply by marchoeppner** [reply]
For low read counts after cutadapt:

https://github.com/bio-raum/FooDMe2/blob/93dbff21367bc30c843b05e2c0e628af086ee90b/subworkflows/illumina_workflow/main.nf#L67

For empty blast reports:
https://github.com/bio-raum/FooDMe2/blob/93dbff21367bc30c843b05e2c0e628af086ee90b/subworkflows/blast_taxonomy/main.nf#L75

In either case, the pipeline will print a temporary message to the screen to let users know that the sample stopped. We can and should probably add this to some sort of audit trail in case the user misses the on-screen error.
